framework/src/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1440,6 +1440,12 @@ public AnnotatedArrayType getErased() {
             at.setComponentType(ct);
             return at;
         }
+
+        @Override
+        public void clearAnnotations() {
```

Comment:
The javadoc for the super method only specifies that the main modifier should be cleared.
This change now also clears array component types. Should this also be done for other types, e.g. `AnnotatedDeclaredType`?
Why is this change necessary? If we agree this is a good change, the javadoc needs to be updated.